src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -973,6 +969,44 @@ public override void DidReceiveChallenge (NSUrlSession session, NSUrlSessionTask
 				}
 			}
 
+			bool InvokeServerCertificateCustomValidationCallback (HttpRequestMessage request, SecTrust secTrust)
+			{
+				if (sessionHandler.ServerCertificateCustomValidationCallback is null)
+					return false;
+
+				var originalChain = secTrust.GetCertificateChain (); // TODO does this work for older iOS and mac versions?
```

Comment:
It does not, look at the bindings:
```csharp
#if NET
		[SupportedOSPlatform ("tvos15.0")]
		[SupportedOSPlatform ("macos12.0")]
		[SupportedOSPlatform ("ios15.0")]
		[SupportedOSPlatform ("maccatalyst15.0")]
#else
		[Watch (8,0)]
		[TV (15,0)]
		[Mac (12,0)]
		[iOS (15,0)]
		[MacCatalyst (15,0)]
#endif
		public SecCertificate[] GetCertificateChain ()
			=> NSArray.ArrayFromHandle<SecCertificate> (SecTrustCopyCertificateChain (Handle));
```

It is using https://developer.apple.com/documentation/security/3747134-sectrustcopycertificatechain The code will fail on older devices, you need to enter use the alternative, or what we could do is just mark this new method to be only supported on those iOS versions that support the C call. 

@rolfbjarne any thoughts?

Reply:
Looks like this could be implemented in earlier versions using [SecTrustGetCertificateCount ](https://developer.apple.com/documentation/security/1397024-sectrustgetcertificatecount?language=objc) and [SecTrustGetCertificateAtIndex ](https://developer.apple.com/documentation/security/1395987-sectrustgetcertificateatindex?language=objc).